holidays/countries/trinidad_and_tobago.py

```diff
@@ -0,0 +1,163 @@
+    IslamicHolidays,
+)
+from holidays.holiday_base import HolidayBase
+
+
+class TrinidadAndTobago(
+    HolidayBase, ChristianHolidays, HinduCalendarHolidays, InternationalHolidays, IslamicHolidays
+):
+    """Trinidad and Tobago holidays.
+
+    References:
+        * <https://otp.tt/trinidad-and-tobago/national-holidays-and-awards/>
+        * <https://en.wikipedia.org/wiki/Public_holidays_in_Trinidad_and_Tobago>
+        * <https://en.wikipedia.org/wiki/Trinidad_and_Tobago_Carnival>
+        * <https://en.wikipedia.org/wiki/Indian_Arrival_Day#Trinidad_and_Tobago>
+        * <https://www.timeanddate.com/holidays/trinidad/eid-al-fitr>
+        * <https://www.timeanddate.com/holidays/trinidad/diwali>
+    """
+
+    country = "TT"
+    default_language = "en_TT"
+    supported_categories = (OPTIONAL, PUBLIC)
+    # %s (estimated).
+    estimated_label = tr("%s (estimated)")
+    supported_languages = ("en_TT", "en_US")
+    # Trinidad and Tobago gained independence on August 31, 1962.
+    start_year = 1963
+
+    def __init__(self, islamic_show_estimated: bool = True, *args, **kwargs):
+        """
+        Args:
+            islamic_show_estimated:
+                Whether to add "estimated" label to Islamic holidays name
+                if holiday date is estimated.
+        """
+        ChristianHolidays.__init__(self)
+        HinduCalendarHolidays.__init__(self, cls=TrinidadAndTobagoHinduHolidays)
+        InternationalHolidays.__init__(self)
+        IslamicHolidays.__init__(
+            self, cls=TrinidadAndTobagoIslamicHolidays, show_estimated=islamic_show_estimated
+        )
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        # New Year's Day.
+        self._add_new_years_day(tr("New Year's Day"))
+
+        # Good Friday.
+        self._add_good_friday(tr("Good Friday"))
+
+        # Easter Monday.
+        self._add_easter_monday(tr("Easter Monday"))
+
+        if self._year >= 1996:
+            # Spiritual Baptist Liberation Day.
+            self._add_holiday_mar_30(tr("Spiritual Baptist Liberation Day"))
+
+            # Indian Arrival Day.
+            self._add_holiday_may_30(tr("Indian Arrival Day"))
+
+        # Corpus Christi.
+        self._add_corpus_christi_day(tr("Corpus Christi"))
+
+        if self._year >= 1973:
+            # Labor Day.
+            self._add_holiday_jun_19(tr("Labour Day"))
+
+        if self._year >= 1985:
+            # African Emancipation Day.
+            self._add_holiday_aug_1(tr("African Emancipation Day"))
+
+        # Independence Day.
+        self._add_holiday_aug_31(tr("Independence Day"))
+
+        if self._year >= 1976:
+            # Republic Day.
+            self._add_holiday_sep_24(tr("Republic Day"))
+
+        # Diwali.
+        self._add_diwali(tr("Diwali"))
+
+        # Christmas Day.
+        self._add_christmas_day(tr("Christmas Day"))
+
+        # Boxing Day.
+        self._add_christmas_day_two(tr("Boxing Day"))
```

Comment:
For consecutive days, `observed_rule` override may be needed here i.e. Christmas ones:
```suggestion
        # Christmas Day.
        dts_observed.add(self._add_christmas_day(tr("Christmas Day")), rule=SUN_TO_NEXT_TUE)

        # Boxing Day.
        dts_observed.add(self._add_christmas_day_two(tr("Boxing Day")))
```

Reply:
```
       # Christmas Day.
>       dts_observed.add(self._add_christmas_day(tr("Christmas Day")), rule=SUN_TO_NEXT_TUE)
E       TypeError: set.add() takes no keyword arguments
```
@PPsyrius I don't think set.add() accepts a second parameter... Could you clarify?

Reply:
Yes, `dts_observed`-way don't supports supports only one rule, so seems we need to use `SUN_TO_NEXT_WORKDAY`. And implement support for _"When two public holidays fall on the same day, the following day is also given as a public holiday"_ separately. I think it will be similar to South Korea.

Reply:
There are two cases we should be considering:
1. When Boxing Day falls on Sunday,
-> Boxing Day is observed on Monday.
2. When Christmas Day falls on Sunday,
-> Christmas Day is observed on Monday. So, Boxing Day is observed on Tuesday.

This means that, Boxing day needs to be shifted from Monday to Tuesday only when Christmas Day falls on Sunday. I think we can do something like this:
`boxing_day_date + timedelta(days=+1)`

So, we can apply SUN_TO_NEXT_MON rule to these holidays:
1. New Years Day
2. Spiritual Baptist Liberation Day
3. Indian Arrival Day
4. Labour Day
5. Emancipation Day
6. Independence Day
7. Republic Day
8. Divali
9. Christmas Day
10. Boxing Day
11. Eid-Ul-Fitr

**Is it correct?**

I have also found that from the year 2000 to 2025, there is one more instance which follows the same pattern:(2025, 3, 30) -> Spiritual Baptist Liberation Day (Sunday)
(2025, 3, 31) -> Eid-Ul-Fitr

Reply:
Also,I think, applying SUN_TO_NEXT_WORKDAY on Christmas Day (Sunday), will shift it to Tuesday as Monday is Boxing Day. I guess, that's not what we want...

Reply:
@KJhellico ?

Reply:
> 2. When Christmas Day falls on Sunday,
>    -> Christmas Day is observed on Monday. So, Boxing Day is observed on Tuesday.

Why exactly _Boxing Day_ is observed on Tuesday? 🙄 I think, it's Christmas Day observed.

Reply:
@KJhellico https://foreign.gov.tt/documents/716/Public_Holidays_2022.pdf

Reply:
If Christmas Day shifts to Monday, then boxing day has to move to Tuesday because "When two public holidays fall on the same day, the following day is also given as a public holiday" as you mentioned.

Reply:
"When two public holidays fall on the same day" - it's like 2025-06-19, when Labour Day and Corpus Christi overlap. Christmas Day/Boxing Day case - it's "next following day that is not a public holiday shall be a public holiday".

Reply:
@KJhellico , These are the years when Christmas Day fell on Sunday:
1. [2005](https://holidayapi.com/countries/tt-pos/2005)
2. [2011](https://holidayapi.com/countries/tt/2011)
3. [2016](https://holidayapi.com/countries/tt/2016)
4. [2022](https://foreign.gov.tt/documents/716/Public_Holidays_2022.pdf)
And so, Christmas is observed on Monday (26th) and Boxing Day is observed on Tuesday (27th)

Reply:
OK, you can try to implement this.

Reply:
@Roshan-1024 @KJhellico I did look a bit into this and only the foreign affairs (Trinidad & Tobago consulate in Panama) labelled as such others left their names empty
- [2016](https://archives.newsday.co.tt/2016/11/19/eca-announces-public-holidays/)
- [2022](https://trinidadexpress.com/newsextra/workers-get-an-additional-public-holiday-for-christmas/article_213c2b4a-61e6-11ed-81ca-0ba4742cf580.html)

Reply:
Hey, @PPsyrius @KJhellico 
Just to clarify, should I still proceed with the implementation where Christmas falling on a Sunday leads to Boxing Day being observed on Tuesday? Or would it be better to keep Boxing Day on Monday unless there's clear overlap, like with Labour Day and Corpus Christi?

Reply:
Is there an official rule that determines which holiday is prioritized when two fall on the same day?

Reply:
> Or would it be better to keep Boxing Day on Monday

As I said, it will be enough to leave Christmas Day and Boxing Day in their usual places, just implement observed holiday.

> Also, in overlapping cases like Labour Day and Corpus Christi, which holiday do we consider as observed on the next day?

Since there is no clear statement on which of the two holidays is transferred, I think it would be better to transfer both (to one day, of course).

> Is there an official rule that determines which holiday is prioritized when two fall on the same day?

I don't see any such rule.

Reply:
Hey @KJhellico , I’ve implemented the overlapping holiday observation logic (e.g., Corpus Christi and Labour Day on 2025-06-19 observed on 2025-06-20). Also, I found that this exact rule doesn’t exist in SK... So, please take a look and share any feedback or changes - I’m open to suggestions.

Reply:
Exact rule - of course, not. But the concept itself...